src/adcm_pytest_plugin/docker/launchers.py

```diff
@@ -0,0 +1,298 @@
+                    step(self)
+
+        if self._local_repository in self.image_name:
+            with allure.step("Remove prepared ADCM image"):
+                for container in self._docker.containers.list(filters={"ancestor": self.image_name}):
+                    with suppress_docker_wait_error():
+                        container.wait(condition="removed", timeout=30)
+
+                with suppress(ImageNotFound):
+                    retry(self._docker.images.remove, self.image_name, force=True, wait_between_=2.0)
+
+    def add_step_fact(self, key: str, value: Any) -> None:
+        self._stages_info[key] = value
+
+    def get_step_fact(self, key: str) -> Optional[Any]:
+        return self._stages_info.get(key)
+
+
+class ADCMWithPostgresLauncher(ADCMLauncher):
+
+    _adcm_class = ADCMWithPostgres
+
+    def __init__(self, adcm_image: Tuple[str, str], docker_client: DockerClient, bind_ip: str, stages: Stages, **_):
+        super().__init__(adcm_image=adcm_image, docker_client=docker_client, bind_ip=bind_ip, stages=stages)
+        self.postgres: Optional[PostgreSQL] = None
+
+    def _run_postgres(self):
+        repo, tag = "postgres", "latest"
+
+        postgres_image = self._docker.images.pull(repository=repo, tag=tag)
+
+        name = f"db-{random_string(6)}"
+
+        with allure.step(f"Launch container with Postgres {tag} from repository {repo}"):
+            container: Container = self._docker.containers.run(
+                image=postgres_image.id,
+                name=name,
+                environment={
+                    "POSTGRES_PASSWORD": "postgres",
+                    "POSTGRES_ADCM_PASS": "password",
+                },
+                network="bridge",
+                remove=True,
+                detach=True,
+            )
+
+        with allure.step("Wait until container is started properly"):
+            wait_kwargs = {"attempts_": 30, "wait_between_": 0.2, "err_type_": RuntimeError}
+            with allure.step("Wait DB is up"):
+                retry(
+                    lambda: "database system is ready to accept connections" in container.logs(tail=15).decode("utf-8"),
+                    err_message_=lambda: f"Failed to start container:\n{container.logs().decode('utf-8').splitlines()}",
+                    **wait_kwargs,
+                )
+            with allure.step("Wait psql"):
+                retry(
+                    lambda: container.exec_run(["psql", "--username", "postgres", "-c", "\\dt"]).exit_code == 0,
+                    err_message_=lambda: f"psql is not available\n{container.logs().decode('utf-8').splitlines()}",
+                    **wait_kwargs,
+                )
+
+        with allure.step("Prepare database"):
+            for statement in (
+                "CREATE USER adcm WITH ENCRYPTED PASSWORD 'password';",
+                "CREATE DATABASE adcm OWNER adcm;",
+                "ALTER USER adcm CREATEDB;",
```

Comment:
Delete role CREATEDB, bc adcm-user has enough grants for adcm-db

Reply:
done